app/src/features/surveys/view/survey-animals/SurveyAnimalsTableActions.tsx

```diff
@@ -0,0 +1,114 @@
+import { mdiDotsVertical, mdiPencilOutline, mdiPlus, mdiTrashCanOutline } from '@mdi/js';
+import Icon from '@mdi/react';
+import IconButton from '@mui/material/IconButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import Typography from '@mui/material/Typography';
+import { useState } from 'react';
+import { IAnimalDeployment } from './animal';
+
+export interface ITableActionsMenuProps {
+  critter_id: number;
+  devices?: IAnimalDeployment[];
+  onAddDevice: (critter_id: number) => void;
+  onRemoveDevice: (critter_id: number) => void;
+  onEditDevice: (critter_id: number) => void;
+  onEditCritter: (critter_id: number) => void;
+  onRemoveCritter: (critter_id: number) => void;
+}
+
+const SurveyAnimalsTableActions = (props: ITableActionsMenuProps) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+
+  const handleClick = (event: any) => {
```

Comment:
Nit: give types to event objects. You can hover over the `onClick` prop on the component and the IDE will reveal what the type is